packages/cdktf/lib/terraform-stack.ts

```diff
@@ -11,6 +11,26 @@ import { Manifest } from './manifest'
 
 const STACK_SYMBOL = Symbol.for('ckdtf/TerraformStack');
 
+
+/**
+ * Returns the naming scheme used to allocate logical IDs. By default, uses
+ * the `HashedAddressingScheme` but this method can be overridden to customize
+ * this behavior.
+ *
+ * @param tfElement The element for which the logical ID is allocated.
+ */
+export function allocateLogicalId(node: Node, stack: TerraformStack): string {
```

Comment:
By changing this to a function, I'm no longer sure that users can actually override the strategy being used.

Reply:
How would a user override this and why? (Sorry I lack context here 🙈 )

Would this solve the problem?

```ts
export class TerraformAsset extends Resource {

  protected allocateLogicalId = allocateLogicalId;
```

Reply:
I understand your comment now, will try to use stack.getLogicalID in the asset :)